### PR TITLE
enable/disable all html-tags corresponding to a radio button checked state

### PR DESCRIPTION
**Pull request by ximex**
fixes parts of https://github.com/engelsystem/engelsystem/issues/1041

missing:
set the `data-radio-name="xxx"` and `data-radio-value="yyy"` attributes on the html tags that should update its disabled state
possible html tags: 'button', 'command', 'fieldset', 'input', 'keygen', 'optgroup', 'option', 'select', 'textarea'

already tested on the shift-creation page with adding the tags for testing only

### CODE REVIEW

**Comment by ximex** [comment]
the shift-creation page is the only one that needs this. added the data attributes for the 2 radio button groups

File: includes/pages/admin_shifts.php [hunk 1]
```diff
@@ -422,7 +422,8 @@ function admin_shifts()
             . form_spinner(
                 'angeltype_count_' . $type->id,
                 $type->name,
-                $needed_angel_types[$type->id]
+                $needed_angel_types[$type->id],
+                ['data-radio-name="angelmode"', 'data-radio-value="manually"']
```
Comment (frischler) [inline includes/pages/admin_shifts.php:426]: I would prefer to have an object here instead of the list, similar to how the "opt" parameter is used in all the macros in form.twig.
So, it would look like this:
{'data-radio-name': 'angelmode', 'data-radio-value': 'manually'}

Comment (ximex) [reply includes/pages/admin_shifts.php]: changed

Comment (MyIgel) [reply includes/pages/admin_shifts.php]: Eventually (as this are only data attributes) explicitly save define them as that? So that its more `['radio-name' => 'angelmode']`

Comment (ximex) [reply includes/pages/admin_shifts.php]: good idea. changed

**Comment by MyIgel** [comment]
Looks good so far to me, @weeman1337 do you have any additional requests?

File: resources/assets/js/forms.js [hunk 2]
```diff
@@ -111,6 +111,35 @@ ready(() => {
   });
 });
 
+/**
+ * @see https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/disabled
```
Comment (weeman1337) [inline resources/assets/js/forms.js:115]:
```suggestion
 * {@link https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/disabled#overview}
```

File: resources/assets/js/forms.js [hunk 3]
```diff
@@ -111,6 +111,35 @@ ready(() => {
...
+ */
+const DISABLE_ELEMENTS = [
+  'button', 'command', 'fieldset', 'input', 'keygen', 'optgroup', 'option', 'select', 'textarea'
+];
+ready(() => {
+  document.querySelectorAll('input[type="radio"]').forEach((radioElement) => {
```
Comment (weeman1337) [inline resources/assets/js/forms.js:121]: Can you please add some description about what this entire block does? It is really hard to follow…

File: includes/sys_form.php [hunk 4]
```diff
@@ -21,19 +21,24 @@ function form_hidden($name, $value)
  * @param string $name
  * @param string $label
  * @param int    $value
+ * @param array  $data_attributes
```
Comment (weeman1337) [inline includes/sys_form.php:24]: Even we are in `includes` this is new code. Can you specify the array contents here please?
https://phpstan.org/writing-php-code/phpdoc-types

**Review by weeman1337 — CHANGES_REQUESTED** [review]
Can you have a look at the comments @ximex . Still needs a round of testing in the application by me.

File: resources/assets/js/forms.js (hunk 2 above)
Comment (ximex) [reply resources/assets/js/forms.js]: changed

File: resources/assets/js/forms.js (hunk 3 above)
Comment (ximex) [reply resources/assets/js/forms.js:122]: added more comments

File: includes/sys_form.php (hunk 4 above)
Comment (ximex) [reply includes/sys_form.php:24]: what do you mean? instead of `array` use `string[]` or `array<string, string>`?

**Comment by ximex** [comment]
also positiv with this solution: if you have disabled javascript it still works

File: includes/sys_form.php (hunk 4 above)
Comment (MyIgel) [reply includes/sys_form.php:24]: More the later one i think ^^

**Review by weeman1337 — APPROVED** [review]
